File: Python/VelocidadGestores3/src/velocidadgestores/ContadorTiempo.py
import datetime


class ContadorTiempo:
    def __init__(self):
        self.tiempo_inicio = datetime.datetime.now()
        self.tiempo_fin = self.tiempo_inicio + datetime.timedelta(minutes=1)

    def tiempo_expirado(self):
        """Verifica si el minuto ha pasado"""
        return datetime.datetime.now() >= self.tiempo_fin

    # def __init__(self):
    #     self.tiempo = datetime.datetime.now()
    #     self.hora_ahora = self.tiempo.hour
    #     self.minuto_ahora = self.tiempo.minute
    #     self.segundo_ahora = self.tiempo.second
    #
    #     self.tiempo = self.tiempo + datetime.timedelta(minutes=1)
    #
    #     self.hora_despues = self.tiempo.hour
    #     self.minuto_despues = self.tiempo.minute
    #     self.segundo_despues = self.tiempo.second
    #
    # def print_valores(self):
    #     print("Hora actual:", self.hora_ahora)
    #     print("Minuto actual:", self.minuto_ahora)
    #     print("Segundo actual:", self.segundo_ahora)
    #     print("Hora después:", self.hora_despues)
    #     print("Minuto después:", self.minuto_despues)
    #     print("Segundo después:", self.segundo_despues)

File: src/main/java/velocidadgestores/utils/ContadorTiempo.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package velocidadgestores.utils;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

import java.util.Calendar;

/**
 *
 * @author MCROBERTW
 */

public class ContadorTiempo {
	private LocalDateTime ahora;
	private LocalDateTime unMinutoDespues;

	// Constructor para inicializar los tiempos
	public ContadorTiempo() {
		this.ahora = LocalDateTime.now();
		this.unMinutoDespues = ahora.plusMinutes(1);
	}

	// Métodos para obtener la hora, minuto y segundo actuales
	public int getHoraAhora() {
		return ahora.getHour();
	}

	public int getMinutoAhora() {
		return ahora.getMinute();
	}

	public int getSegundoAhora() {
		return ahora.getSecond();
	}

	// Métodos para obtener la hora, minuto y segundo después de un minuto
	public int getHoraDespues() {
		return unMinutoDespues.getHour();
	}

	public int getMinutoDespues() {
		return unMinutoDespues.getMinute();
	}

	public int getSegundoDespues() {
		return unMinutoDespues.getSecond();
	}

	// Método para formatear el tiempo en una cadena de texto
	public String getFormattedTiempoAhora() {
		return ahora.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
	}

	public String getFormattedTiempoDespues() {
		return unMinutoDespues.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
	}
}


/*public class ContadorTiempo {
    int horaAhora, minutoAhora, segundoAhora;
	int horaDespues, minutoDespues, segundoDespues;
	Calendar tiempo =Calendar.getInstance();


	public void ContadorTiempo()
	{

		horaAhora=tiempo.get(Calendar.HOUR);
		minutoAhora=tiempo.get(Calendar.MINUTE);
		segundoAhora=tiempo.get(Calendar.SECOND);

		tiempo.roll(Calendar.MINUTE, true);

		horaDespues=tiempo.get(Calendar.HOUR);
		minutoDespues=tiempo.get(Calendar.MINUTE);
		segundoDespues=tiempo.get(Calendar.SECOND);

		*//*System.out.println(horaAhora);
		System.out.println(minutoAhora);
		System.out.println(segundoAhora);

		System.out.println(horaDespues);
		System.out.println(minutoDespues);
		System.out.println(segundoDespues);*//*
}
}*/
